inkcpp/include/functional.h

```diff
@@ -74,7 +74,7 @@ namespace ink::runtime::internal
 			static_assert(sizeof...(Is) == traits::arity);
 
 			// void functions
-			if constexpr (is_same<void, traits::return_type>::value)
+			if constexpr (is_same<void, typename traits::return_type>::value)
```

Comment:
The compiler doesn't know that `return_type` in `traits` is a type, so we need the `typename` qualifier.